### ISSUE TITLE
Availability of Console for apk

### ISSUE THREAD

**Issue by somewhatnmc** [spesmilo/electrum#5375]
Unsure if WAD but should the Console be available on Android? It would be swell if it was available.

**Comment by bauerj** [comment]
What exactly are you trying to achieve with the console?

**Comment by somewhatnmc** [comment]
RPC calls.

**Comment by bauerj** [comment]
What exactly are you trying to achieve by using RPC calls?

**Comment by somewhatnmc** [comment]
My immediate concern is I just installed downstream electrum-nmc and would like to use RPC calls to renew names via console. With that said, console is available for Electrum with the exception of the apk so I am wondering if it can be made available.

**Comment by JeremyRand** [comment]
There are a couple of different use cases here, and it may make sense for them to split off into separate issues.

The first use case is accessing a GUI for the debug console in the Kivy GUI.  This is useful mainly for debugging purposes and power-user purposes (i.e. the same reason it's useful in the Qt GUI).  In Namecoin's case, power-user purposes would happen to include name wallet operations (because there's no Kivy GUI for that yet), but there's nothing Namecoin-specific about the console being useful for power users and debugging, nor would Namecoin's need for this go away after Namecoin adds a Kivy GUI for name operations (since the console API is a lot more flexible than either the Qt or Kivy GUI's, in both Bitcoin and Namecoin).

The second use case is exposing an RPC API to other applications.  This is useful for other applications that want to access the blockchain state and/or perform wallet operations (same reason it's useful on the desktop).  In Namecoin's case, we're intending to have a Namecoin-to-DNS bridge app on Android that exposes a DNS server, whose results are determined by querying Electrum's RPC port.  Again though, this is not a Namecoin-specific use case -- there are a wide variety of reasons why another app might want to query the blockchain and/or issue wallet operations, and the motivations here are basically the same whether it's the desktop app or the Android app.

Hope this helps clarify why the Namecoin community is interested in this stuff, and also clarifies why we'd like to see this kind of functionality happen upstream.

**Comment by ecdsa** [comment]
I think this would be a waste of our time
1. I fail to see why accessing the console in Kivy is useful. 
2. exposing RPC to other applications does not require a console

**Comment by SomberNight** [comment]
I too find it hard to come up with use cases, but there are probably some.
Debugging is a reasonable example -- not sure how much it would be used on Kivy...
In any case, clearly this is for advanced users only, so it should be hidden by default.

I would say If someone contributes this, it could get merged -- but try to keep it simple to minimise maintenance costs: the Qt console has some bells and whistles that would probably be overkill for Kivy.

Note that the current console implementation for Qt (`gui.qt.console`) is really specific to Qt, so a console for Kivy would need to be written from ~scratch.

**Comment by JeremyRand** [comment]
Yes, to be clear: I'm not expecting this to be a high priority for you guys; I'm just requesting that it be merged if someone (maybe one of the Namecoin people) contributes the code and it passes review.  We're not interested in being a burden on you, we just want to make sure that the stuff we create ends up benefiting upstream too.

**Comment by benma** [comment]
I am helping a user get some funds unstuck that are buried beyond the regular gap limit and wanted them to increase the gap limit via the console (I don't think there is another way?). The user is a mobile-only user, so this would have been helpful.

Just putting this here for the record, I agree though that it is a low priority feature.
